contracts/src/IVerifier.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.17;
+
+/**
+ * @title Interface of Verifier contract
+ * @author Seiya Kobayashi
+ */
+interface IVerifier {
+    /// @dev Holds hash value of node
+    struct Node {
+        bytes32 value;
+    }
+
+    /// @dev Holds hash value of the matched node and hash values of its Merkle path (to calculate root hash)
+    struct MatchedNode {
```

Comment:
I think this is so-called merkle inclusion proof.

Reply:
Renamed struct names in c4e11aa44544d4c596f6a3a74d7db945bbd4b8c7.
Looking into [this library](https://github.com/SeiyaKobayashi/traininig-optimization-on-ml-model-with-zkp/pull/5#discussion_r1092655698) to simplify Merkle tree implementation.